src/main/webapp/app/overview/exercise-details/course-exercise-details.component.ts

```diff
@@ -81,6 +79,7 @@ export class CourseExerciseDetailsComponent implements OnInit, OnDestroy {
         } else {
             this.exerciseService.findResultsForExercise(this.exerciseId).subscribe((exerciseResponse: HttpResponse<Exercise>) => {
                 this.exercise = exerciseResponse.body;
+                this.exercise.participations = this.filterParticipations(this.exercise.participations);
```

Comment:
The result of the method filterParticipations is used for an assignment, however sortParticipationsFinishedFirst which is called within this method, mutates this.exercise.participations by calling sort on it (Array.prototype.sort sorts in place: https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Array/sort).
This can lead to unexpected bugs, if e.g. somebody reuses the sortParticipations method for different purposes.
In my opinion it would be better to use lodash sort, which returns a new array: https://lodash.com/docs/#sortBy

Reply:
i think it is ok how we are using it, because we directly assign the sorting to the exercise model on the component. if we would return the array i would agree

Reply:
It was not intended that `sortParticipationsFinishedFirst` operates on `this.exercise.participations`. I changed it to use the participation array passed as argument. Otherwise, it would have no impact in line 82 here, as the sort function sorts `this.exercise.participations` and afterwards the array returned from the `filterParticipations` function is assigned to it, without being sorted. 

Now I think it's fine. I also added a note to the sort function saying that it operates directly on the argument and does not return anything.